src/Data/RFC3339String.purs

```diff
@@ -41,7 +41,10 @@ trim (RFC3339String s) =
     withoutTrailingZeros = dropWhileEnd (eq '0') withoutZulu
     withoutTrailingDot = dropWhileEnd (eq '.') withoutTrailingZeros
   in
-    RFC3339String $ withoutTrailingDot <> "Z"
+    if withoutTrailingZeros == withoutTrailingDot
+    then RFC3339String $ withoutTrailingDot <> "Z"
+         -- always have a subseconds component
+    else RFC3339String $ withoutTrailingDot <> ".0Z"
```

Comment:
@themattchan could you explain the motivation for this?

Reply:
An existing test case in query language expects a subseconds component.